packages/cli/generation/ir-generator/src/casings/CasingsGenerator.ts

```diff
@@ -24,38 +24,48 @@ export function constructCasingsGenerator({
-                camelCaseName = camelCaseWords
-                    .map((word, index) => {
-                        if (index > 0) {
+            if (smartCasing) {
+                if (!hasAdjacentCommonInitialisms(camelCaseWords)) {
+                    camelCaseName = camelCaseWords
+                        .map((word, index) => {
+                            if (index > 0) {
+                                const pluralInitialism = maybeGetPluralInitialism(word);
+                                if (pluralInitialism != null) {
+                                    return pluralInitialism;
+                                }
+                                if (isCommonInitialism(word)) {
+                                    return word.toUpperCase();
+                                }
+                            }
+                            return word;
+                        })
+                        .join("");
+                    pascalCaseName = camelCaseWords
+                        .map((word, index) => {
                             const pluralInitialism = maybeGetPluralInitialism(word);
                             if (pluralInitialism != null) {
                                 return pluralInitialism;
                             }
                             if (isCommonInitialism(word)) {
                                 return word.toUpperCase();
                             }
-                        }
-                        return word;
-                    })
-                    .join("");
-                pascalCaseName = camelCaseWords
-                    .map((word, index) => {
-                        const pluralInitialism = maybeGetPluralInitialism(word);
-                        if (pluralInitialism != null) {
-                            return pluralInitialism;
-                        }
-                        if (isCommonInitialism(word)) {
-                            return word.toUpperCase();
-                        }
-                        if (index === 0) {
-                            return upperFirst(word);
-                        }
-                        return word;
-                    })
-                    .join("");
+                            if (index === 0) {
+                                return upperFirst(word);
+                            }
+                            return word;
+                        })
+                        .join("");
+                }
+
+                // In smartCasing, manage numbers next to letters differently:
+                // _.snakeCase("v2") = "v_2"
+                // smartCasing("v2") = "v2", other examples: "test2This2 2v22" => "test2this2_2v22", "applicationV1" => "application_v1"
+                snakeCaseName = name
+                    .split(" ")
+                    .map((part) => part.split(/(\d+)/).map(snakeCase).join(""))
+                    .join("_");
```

Comment:
@armandobelardo can we add some tests here ?

Reply:
https://github.com/fern-api/fern/blob/main/packages/cli/generation/ir-generator/src/__test__/casingsGenerator.test.ts